### PR TITLE
reduce memory consumption of share extension 

### PR DESCRIPTION
**Pull request by cyBerta**
by scaling down image thumbnail

fixes point one of https://github.com/deltachat/deltachat-ios/issues/1330#issuecomment-918108809

### CODE REVIEW

File: DcShare/Helper/ShareAttachment.swift [hunk 1]
```diff
@@ -120,7 +120,7 @@ class ShareAttachment {
                 self.messages.append(msg)
                 self.delegate?.onAttachmentChanged()
                 if self.imageThumbnail == nil {
-                    self.imageThumbnail = result
+                    self.imageThumbnail = result.scaleDownImage(toMax: 96)
```
Comment (r10s) [inline DcShare/Helper/ShareAttachment.swift:123]: what exactly does this `96` mean? max width/height in pixel? would be nice to have some comment at the corresponding function (not really an issue of this pr).

**Review by r10s — APPROVED** [review]

**Comment by r10s** [comment]
do you know how much memory this frees up?

i did not manage to share a panorama still.

File: DcShare/Helper/ShareAttachment.swift (hunk 1 above)
Comment (cyBerta) [reply DcShare/Helper/ShareAttachment.swift]: I replaced the hard coded value by `self.thumbnailSize` which also helps to understand where the value comes from.

**Comment by cyBerta** [comment]
> do you know how much memory this frees up?
> 
> i did not manage to share a panorama still.

It reduces the amount of mermory invovled for the 9.5mb image from ~70 mb to ~14 mb. However, the image processing in the core still consumes too much memory.

**Pull request merged**